### PR TITLE
Use Otel Collector

### PR DESCRIPTION
**Pull request by louiseschmidtgen**
## Otel Collector
Otel has recently made updates to their [collector](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/examples/otel-collector/README.md), allowing the use of the collector for both metrics and traces and to forward them to jaeger and prometheus.

This code alligns with Otel's example set-up for the collector: https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/examples/otel-collector

### CODE REVIEW

File: pkg/instrument/docker-compose.yaml
Comment (bschimke95) [inline pkg/instrument/docker-compose.yaml:1]: The `pkg` folder should only contain go source code. Maybe `hack` fits better?

**Review by bschimke95 — COMMENTED** [review]
One comment but looks good otherwise.

File: pkg/instrument/docker-compose.yaml
Comment (louiseschmidtgen) [reply pkg/instrument/docker-compose.yaml:1]: True, I'll make a new folder under hack named 'otel' to separate it from all the build scripts.

**Review by bschimke95 — APPROVED** [review]
LGTM! Great work @louiseschmidtgen

**Pull request merged**